main.cpp

```diff
@@ -1,10 +1,58 @@
 #include <iostream>
+#include <iomanip>   // format manipulation
 
 using namespace std;
-
+/**This program displays an equilateral triangle based on the user's input.
+ * The user decides on its size and the character that it may consist of.
+ * The triangle increases by three characters each time and is based on the 
+ * T(n)=2n+1 equation.*/
+void inputCheck(int & size)//Still trying to figure our cin.fail() and cin.good()
+{
+ bool isdigit();
+    if(size.isdigit()==true)
+        
+        cin.good();
+    cout<<endl;
+        
+    else
+        cin.fail();
+        cout << "Please re-enter a size. ";
+        cin.clear();
+        cin>>size;
+}
```

Comment:
The implementation of checking whether the input is valid or not should be the only purpose for this method. Not to request for a re-enter or anything like that -the main method should deal with that. So just ensure this method checks and only checks the input to see it is valid -if it is valid returns true if not returns false. Size needs to be a valid number that can be stored in the int data type and character needs to be a valid value to store within the char data type.